src/main/java/wolox/training/repositories/UserRepository.java

```diff
@@ -16,4 +20,17 @@
      */
     Optional<User> findByUsername(String username);
 
+    /**
+     * Method that obtain all users if your birthdate be between two date
+     *
+     * @param maxDate:        Max date ({@link LocalDate})
+     * @param minDate:        Min date ({@link LocalDate})
+     * @param nameContaining: Name containing of the user (String)
+     * @return {@link User} list
+     */
+    @Query("SELECT u FROM User u WHERE u.birthdate <= :maxDate and u.birthdate >= :minDate and (LOWER(u.name) like LOWER(CONCAT('%', :nameContaining, '%')))")
+    List<User> findByBirthdateBetweenTwoDateCustom(
+        @Param("maxDate") LocalDate maxDate, @Param("minDate") LocalDate minDate,
+        @Param("nameContaining") String nameContaining);
```

Comment:
You can use the IgnoreCase that you have used before, instead of doing it in the query.